keystore.go

```diff
@@ -0,0 +1,27 @@
+package sdk
+
+import (
+	lru "github.com/hashicorp/golang-lru"
```

Comment:
there's a v2 of this package that uses generics. Could we switch to that?

Reply:
I would like to, but it's a little bit awkward because the current definition for JSONWebKeyStore uses `interface{}`:
```
type JSONWebKeyStore interface {
    Get(key interface{}) (value interface{}, ok bool)
    Add(key, value interface{})
}
```

Given that I've already made a breaking change to the API in https://github.com/pomerium/sdk-go/commit/67fafdac317326c6fe97b909c168b727ca2809e9, should we go ahead and specify these interface methods more precisely?

I think what we actually rely on is something more like this:
```
 type JSONWebKeyStore interface {
     Get(keyID string) (value *jose.JSONWebKey, ok bool)
     Add(keyID string, value *jose.JSONWebKey)
 }
```

Then we could go ahead and use the lru v2 package like `lru.New[string, *jose.JSONWebKey]`.

Thoughts?

Reply:
sure

Reply:
Merged https://github.com/pomerium/sdk-go/pull/95 to change the JSONWebKeyStore type (thanks for the quick review!).

Updated this PR to use github.com/hashicorp/golang-lru/v2.